src/main/java/com/wiley/autotest/selenium/elements/upgrade/TeasyElementData.java

```diff
@@ -4,25 +4,27 @@
 import org.openqa.selenium.WebElement;
 
 /**
- * User: ntyukavkin
- * Date: 18.11.2016
- * Time: 16:34
+ * Class-holder for all data that is used fo future TeasyElement creation;
  */
 public class TeasyElementData {
 
+    //element that we found and will be wrapping with TeasyElement;
     private WebElement element;
+    //when we search for an element inside another element;
     private TeasyElement searchContext;
+    //locator for an element that we are searching for;
     private By by;
+    //index of a particular element in a list;
     private Integer index;
-    private Locator locator;
 
-    public TeasyElementData(WebElement element) {
+    public TeasyElementData(WebElement element, By by) {
         this.element = element;
+        this.by = by;
     }
 
-    public TeasyElementData(WebElement element, By by) {
-        this(element);
-        this.by = by;
+    public TeasyElementData(WebElement element, By by, int index) {
```

Comment:
Integer or int?

Reply:
int seems fine here, as for the Constructor we accept a number, but the field is Integert to support a null value. But good point to double check :)